### ISSUE TITLE
Language Server Protocol for Prolog

### ISSUE THREAD

**Issue by erlanger** [SWI-Prolog/swipl-devel#389]
It would be great to implement an LSP  server (language server protocol)  for Prolog, so that we would have code completion, compilation errors, etc support  in many editors, something like this:

![eclipse-css-code-complete](https://user-images.githubusercontent.com/8608855/50314411-a755a180-04a6-11e9-85d0-cc381f04c074.png)
 
but for prolog.

I think someone already did a lot of the work writing a Vscode server for SWI-Prolog (which is essentially an LSP server).

 It would probably be a good idea to use that code to setup a standalone LSP server that would work with any editor. Almost [all languages](https://langserver.org/#implementations-server) have support today, I would not like for Prolog to be left behind.

**Comment by XVilka** [comment]
Seems there is an implementation already: https://github.com/jamesnvc/lsp_server/
But it would be awesome to have it in SWIPL out of the box.

**Comment by erlanger** [comment]
Yes, I tried it, but it doesn't work with neovim's [LanguageClient](https://github.com/autozimu/LanguageClient-neovim).

**Comment by XVilka** [comment]
@erlanger what exactly doesn't work? Maybe it makes sense to open bug in one project or another?
